os/fs/driver/mtd/smart.c

```diff
@@ -2163,6 +2163,44 @@ static int smart_scan(FAR struct smart_struct_s *dev)
 #endif
 			}
 
+			fdbg("Duplicate Sector winner=%d, loser=%d\n", winner, loser);
+
+#ifdef CONFIG_MTD_SMART_ENABLE_CRC
+			/* Check CRC of the winner sector just in case */
+
+			ret = MTD_BREAD(dev->mtd, winner * dev->mtdBlksPerSector, dev->mtdBlksPerSector, (FAR uint8_t *)dev->rwbuffer);
+			if (ret == dev->mtdBlksPerSector) {
```

Comment:
What if "ret != OK", i.e. the read request fails?
We should return an appropriate ERROR code here

Otherwise, continue with the functioning.

Reply:
I agree with @amogh-samsung, if ret is not same as mtdBlksPerSector, error happend in flash driver, and it will return errno.
```
if (ret != dev->mtdBlksPerSector) {
    return ret;
}

ret = smart_validate_crc(dev);
if (ret != OK) {
   ....
}
```

Reply:
I changed